my-app/src/main/java/com/JavaWebServer/app/Decoder.java

```diff
@@ -0,0 +1,19 @@
+package com.JavaWebServer.app;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class Decoder {
```

Comment:
Why does this branch include the decoder? Didn't you cherry pick this branch off of the params branch?